File: bulk/javascript/redirect_01.js
// CWE-601: Open Redirect
const express = require('express');
const app = express();
app.get('/login', (req, res) => {
  // authenticate...
  const nextUrl = req.query.next;
  if (nextUrl && nextUrl.startsWith('/') && !nextUrl.startsWith('//')) {
    res.redirect(nextUrl);
  } else {
    res.redirect('/');
  }
});

File: bulk/go/redirect_01.go
// CWE-601: Open Redirect
package main
import "net/http"
func loginHandler(w http.ResponseWriter, r *http.Request) {
	next := r.URL.Query().Get("next")
	if next == "" {
		next = "/"
	}
	http.Redirect(w, r, next, http.StatusFound)
}
